scipy/optimize/tests/test__optimize.py

```diff
@@ -0,0 +1,47 @@
+import numpy as np
+
+import pytest
+from numpy.testing import assert_allclose, assert_
+
+from scipy.optimize.optimize import MemoizeJac
+
+class FunctionWithGradient(object):
+    def __init__(self):
+        self.number_of_calls = 0
+
+    def __call__(self, x):
+        self.number_of_calls += 1
+        return np.sum(x**2), 2*x
+
+
+@pytest.fixture
+def function_with_gradient():
+    return FunctionWithGradient()
+
+
+def test_memoize_jac_function_before_gradient(function_with_gradient):
+    memoized_function = MemoizeJac(function_with_gradient)
+
+    x0 = np.array([1.0, 2.0])
+    assert_allclose(memoized_function(x0), 5.0)
+    assert_(function_with_gradient.number_of_calls == 1)
+
+    assert_allclose(memoized_function.derivative(x0), 2*x0)
```

Comment:
PEP8 spaces around operator, `2 * x0`.

Reply:
everything should be PEP8 compliant now.